src/main/java/com/google/jenkins/plugins/k8sengine/KubernetesEngineBuilder.java

```diff
@@ -284,6 +293,48 @@ private boolean verify(
         kubectl, manifestObjects, consoleLogger, verifyTimeoutInMinutes);
   }
 
+  /**
+   * @param credentialsId The service account credential's id.
+   * @return Access token from OAuth to allow kubectl to interact with the cluster.
+   * @throws IOException If an error occurred fetching the access token.
+   */
+  public static String getAccessToken(String credentialsId) throws IOException {
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(credentialsId));
+    HttpTransport HTTP_TRANSPORT = new NetHttpTransport();
+    JsonFactory JSON_FACTORY = new JacksonFactory();
+
+    // get the credential from jenkins
+    List<DomainRequirement> domainRequirements = new ArrayList<>();
+
+    GoogleRobotPrivateKeyCredentials cred =
```

Comment:
Much of this logic is duplicated from the ClientFactory. Would prefer to consolidate it centrally.

Reply:
I'm not sure I understand what you mean. Do you mean credentials related methods should be kept in ClientFactory? The only similar code I see is getting the GoogleRobotPrivateKeyCredentials portion. Do you want said portion as a helper in ClientFactory?

Reply:
I'll leave the decision of where the logic leaves to your discretion. My main goal is to eliminate the duplication.

Reply:
Hmm, I see what you're saying now. Yea, think this could be a good de-duplication effort. Thanks

Reply:
Done.